Add core status widget

sebastian-meyer opened the pull request:
This pull request adds a small status message for Solr cores. When editing a `tx_dlf_solrcores` record in the backend, the status of the respective core is fetched from Solr and shown as flash message.

This is done by implementing a new `FieldInformation` renderType for input fields.

---

In addition this pull request refactors the recently introduced warning message for changing the `index_name` of records (#553) from `FieldWizard` to `FieldInformation` as well. Originally this was implemented as `FieldWizard` because `FieldInformation` doesn't allow all HTML tags of a flash message. But in fact, only the `<h4>` tag of the flash message's title is disallowed, so by just omitting the title we can still use flash messages as `FieldInformation`.
Since a static warning message should logically be a `FieldInformation` I decided to refactor this.

sebastian-meyer committed:
Add Core status message

sebastian-meyer committed:
Change FieldWizard to FieldInformation

sebastian-meyer committed:
Fix namespaces

sebastian-meyer committed:
Remove flash message title

sebastian-meyer committed:
Remove flash message title

sebastian-meyer committed:
Remove HSC

sebastian-meyer committed:
Change field label

albig commented on Classes/Hooks/Form/FieldInformation/SolrCoreStatus.php:
```diff
@@ -0,0 +1,78 @@
+                $uptimeInSeconds = floor($response->getUptime() / 1000);
+                $uptime = floor($uptimeInSeconds / 3600) . gmdate(":i:s.", $uptimeInSeconds % 3600) . $response->getUptime() % 1000;
+                $numDocuments = $response->getNumberOfDocuments();
+                $startTime = $response->getStartTime() ? date_format($response->getStartTime(), 'c') : 'N/A';
```
"Start Time" is confusing in my opinion. I would prefer "Creation Time" because it''s the creation time of the core. Not the start time of the Solr. instance.

sebastian-meyer replied:
Actually `startTime` is the time the core was loaded. This can be the creation date or any later time when a core reload or Solr restart happened.
Thus `startTime` and `Uptime` are redundant: `Uptime` is just `NOW - startTime`.

albig commented on Classes/Hooks/Form/FieldInformation/SolrCoreStatus.php:
```diff
@@ -0,0 +1,78 @@
+                $action->setCore($core);
+                $coreAdminQuery->setAction($action);
+                $response = $solr->service->coreAdmin($coreAdminQuery)->getStatusResult();
+                $uptimeInSeconds = floor($response->getUptime() / 1000);
```
Would be easier to read if adding the amount of days. E.g. DD HH:MM:SS.

I think, this  is what the Solr-Core Admin-Panel writes as "startTime: 16 days ago".

Uptime is better than "Start Time". But The "Star Time" below seems to be the creation time. I'm a little confused.

sebastian-meyer replied:
I agree that it would be nice to have the days as well, but it's complicated to calculate. There are so many ways and each has its disadvantages: https://stackoverflow.com/questions/8273804/convert-seconds-into-days-hours-minutes-and-seconds
I think it's just not worth it...

albig reviewed (changes requested):
It's working. Thank you and it's helpful.

The given times are a little confusing. Please check again.

This is what TYPO3 tells me:
	
```
Start Time: 2021-01-20T14:50:16+00:00
Uptime: 22:08:43.511
Last Modified: 2021-01-20T15:35:42+00:00
Number of Documents: 2832
```

This is, what Solr tells me about the Core:

```
startTime:about 22 hours ago
lastModified:about 22 hours ago
Num Docs:2832
```

sebastian-meyer committed:
Generate localized dates

sebastian-meyer commented:
I changed the date output to be localized in order to make it less confusing. But I suggest keeping the `Uptime` as it is.

Commit:
Proposal: Show uptime in format like "uptime" on linux console.

uptime on Linux returns:

  23:41:18 up 4 days,  6:37,  1 user,  load average: 0,91, 0,99, 1,10

The first is the current time, followed by "up", the amount of days
which might be more than a year, followed by h:m.

The proposed output for Solr-uptime would be now:

  Uptime: 0 days, 00:22:56

This includes the seconds, too.

The solution before was:

  Uptime: 0:21:59.385

Which I find harder to read and not self-explaining.

Commit:
Merge branch 'master' into add-core-status

sebastian-meyer committed:
Removed unused variables

sebastian-meyer committed:
Merge pull request #19 from albig/sebastian-meyer-add-core-status

Proposal: Show uptime in format like "uptime" on linux console.

sebastian-meyer commented:
Thanks for your proposal to use the Linux uptime format. I merged your pull request.

sebastian-meyer committed:
Merge branch 'master' into add-core-status
